game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/retreat/OffensiveGeneralRetreat.java

```diff
@@ -105,18 +106,13 @@ private boolean canAttackerRetreatSeaPlanes() {
   }
 
   private boolean canAttackerRetreatPartialAmphib() {
-    if (Properties.getPartialAmphibiousRetreat(battleState.getGameData())) {
-      // Only include land units when checking for allow amphibious retreat
-      final List<Unit> landUnits =
-          CollectionUtils.getMatches(
-              battleState.getUnits(BattleState.Side.OFFENSE), Matches.unitIsLand());
-      for (final Unit unit : landUnits) {
-        if (!unit.getWasAmphibious()) {
-          return true;
-        }
-      }
+    if (!Properties.getPartialAmphibiousRetreat(battleState.getGameData())) {
+      return false;
     }
-    return false;
+    // Only include land units when checking for allow amphibious retreat
+    return battleState.getUnits(BattleState.Side.OFFENSE).stream()
+        .filter(Matches.unitIsLand())
+        .anyMatch(unit -> !unit.getWasAmphibious());
```

Comment:
nitpicky, note - you can use Predicates here to write this as a method references:
```
.anyMatch(Predicates.not(Unit::getWasAmphibious));
```